controller/script/simulate-proxy/main.go

```diff
@@ -119,8 +133,117 @@ var (
 	}
 )
 
-func randomPort() uint32 {
-	return ports[rand.Intn(len(ports))]
+// generateProxyTraffic randomly creates metrics under the guise of a single conduit proxy routing traffic.
+// metrics are generated for each proxyMetricCollector.
+func (s *simulatedProxy) generateProxyTraffic(podList []*v1.Pod) {
+	proxyPodName := strings.Split(s.pod.GetName(), "-")[0]
+
+	for {
+		inboundRandomCount := int(rand.Float64() * 10)
+		outboundRandomCount := int(rand.Float64() * 10)
+		randomDestinationPod := podList[rand.Intn(len(podList))]
+		randomDestinationPodName := strings.Split(randomDestinationPod.GetName(), "-")[0]
+
+		s.requestTotals.
+			With(overrideDefaultLabels(s.newConduitLabel(proxyPodName, randomDestinationPodName))).
+			Add(float64(inboundRandomCount))
+
+		s.responseTotals.
+			With(overrideDefaultLabels(s.newConduitLabel(proxyPodName, randomDestinationPodName))).
+			Add(float64(outboundRandomCount))
+
+		observeHistogramVec(
+			randomLatencies(randomCount()),
+			s.responseLatencyMs,
+			overrideDefaultLabels(s.newConduitLabel(proxyPodName, randomDestinationPodName)))
+
+		observeHistogramVec(
+			randomLatencies(randomCount()),
+			s.requestDurationMs,
+			overrideDefaultLabels(s.newConduitLabel(proxyPodName, randomDestinationPodName)))
+
+		time.Sleep(s.sleep)
+	}
+}
+
+// newConduitLabel creates a label map to be used with metric generation.
+func (s *simulatedProxy) newConduitLabel(proxyPodName string, destinationPod string) prom.Labels {
+	labelMap := prom.Labels{
+		"direction":  randomRequestDirection(),
+		"deployment": proxyPodName,
+		"authority":  "world.greeting:7778",
+		"namespace":  s.namespace,
+	}
+	if labelMap["direction"] == "outbound" {
+		labelMap["dst_deployment"] = destinationPod
+	}
+
+	if rand.Intn(2) == 0 {
+		labelMap["grpc_status_code"] = fmt.Sprintf("%d", randomGrpcResponseCode())
+	} else {
+		labelMap["status_code"] = fmt.Sprintf("%d", randomHttpResponseCode())
+	}
+	return labelMap
+}
+
+func observeHistogramVec(latencyBuckets []uint32, latencies *prom.HistogramVec, latencyLabels prom.Labels) {
+	for bucketNum, count := range latencyBuckets {
+		latencyMs := float64(latencyBucketBounds[bucketNum]) / 10
```

Comment:
this latencyMs could fall into any bucket from `0...bucketNum`.  I think it would be fine to report this as `latencyBucketBounds[bucketNum]/10` (i.e. report all observations at the top of their bucket)